test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/TrxLoggerTests.cs

```diff
@@ -634,6 +635,131 @@ public void DefaultTrxFileShouldCreateIfLogFileNameParameterNotPassed()
+            {
+                if (files != null)
+                {
+                    foreach (var file in files)
+                    {
+                        if (!string.IsNullOrEmpty(file) && File.Exists(file))
+                        {
+                            File.Delete(file);
+                        }
+                    }
+                }
+            }
+        }
+
+        [TestMethod]
+        public void TrxFileNameShouldNotIterate()
+        {
+            var loggers = new TestableTrxLogger[10];
+            for (int i = 0; i < loggers.Length; i++)
+            {
+                loggers[i] = new TestableTrxLogger();
+                loggers[i].Initialize(this.events.Object, this.parameters);
+            }
+
+            string[] files = null;
+            try
+            {
+                loggers.AsParallel().ForAll(l =>
+                {
+                    var pass = TrxLoggerTests.CreatePassTestResultEventArgsMock();
+                    l.TestResultHandler(new object(), pass.Object);
+                    var testRunCompleteEventArgs = TrxLoggerTests.CreateTestRunCompleteEventArgs();
+                    l.TestRunCompleteHandler(new object(), testRunCompleteEventArgs);
+                });
+
+                files = loggers.Select(l => l.trxFile).Distinct().ToArray();
+
+                Assert.IsTrue(files.Length == 1, "All logger instances should get the same file name!");
+            }
+            finally
+            {
+                if (files != null)
+                {
+                    foreach (var file in files)
+                    {
+                        if (!string.IsNullOrEmpty(file) && File.Exists(file))
+                        {
+                            File.Delete(file);
+                        }
+                    }
+                }
+            }
+        }
+
+        [TestMethod]
+        public void TrxPrefixFileNameShouldIterate()
+        {
+            this.parameters.Remove(TrxLoggerConstants.LogFileNameKey);
+            this.parameters[TrxLoggerConstants.LogFilePrefixKey] = DefaultLogFilePrefixParameterValue;
+
+            var loggers = new TestableTrxLogger[10];
+            for (int i = 0; i < loggers.Length; i++)
+            {
+                loggers[i] = new TestableTrxLogger();
+                loggers[i].Initialize(this.events.Object, this.parameters);
+            }
+
+            string[] files = null;
+            try
+            {
+                loggers.AsParallel().ForAll(l =>
+                {
+                    var pass = TrxLoggerTests.CreatePassTestResultEventArgsMock();
+                    l.TestResultHandler(new object(), pass.Object);
+                    var testRunCompleteEventArgs = TrxLoggerTests.CreateTestRunCompleteEventArgs();
+                    l.TestRunCompleteHandler(new object(), testRunCompleteEventArgs);
+                });
+
+                files = loggers.Select(l => l.trxFile).Distinct().ToArray();
```

Comment:
This part is the same in test methods, it is better to have it in separate method.